src/vsync.js

```diff
@@ -51,12 +71,36 @@ export class Vsync {
    * @param {!Object<string, *>|undefined} opt_state
    */
   run(task, opt_state) {
+    // Do not request animation frames when the document is not visible.
+    if (!viewerFor(this.win).isVisible()) {
+      return;
+    }
     const state = opt_state || {};
-    this.win.requestAnimationFrame(() => {
-      if (task.measure) {
-        task.measure(state);
+    this.tasks_.push(task);
+    this.states_.push(state);
+
+    if (this.scheduled_) {
+      return;
+    }
+    this.scheduled_ = true;
+
+    this.raf_(() => {
+      this.scheduled_ = false;
+      // TODO(malteubl) Avoid array allocation with a double buffer.
+      const tasks = this.tasks_;
+      const states = this.states_;
+      this.tasks_ = [];
+      this.states_ = [];
+      for (let i = 0; i < tasks.length; i++) {
+        if (tasks[i].measure) {
+          tasks[i].measure(states[i]);
+        }
+      }
+      for (let i = 0; i < tasks.length; i++) {
+        if (tasks[i].mutate) {
```

Comment:
Could you please modify the definition of `mutate` above in `VsyncTask` struct as optional?